### PR TITLE
Filtering

### PR DESCRIPTION
**Pull request by PrashanthKumarM**
3 filters implemented. The ones based on list options (documentRouteStatus, documentType and actionRequested)

### CODE REVIEW

File: actionlist/app/components/SideMenu.js [hunk 1]
```diff
@@ -48,54 +78,98 @@ const style = StyleSheet.create({
...
   optionSelected: state.actionItemsReducer.optionSelected,
+  filters: state.actionItemsReducer.filterStatus,
 });
 
 const mapDispatchToProps = dispatch => ({
+  onReset: () => dispatch(resetFilters()),
   onSort: criteria => dispatch(sortActionList(criteria)),
 });
 
+/* eslint-enable no-unused-vars */
```
Comment (Nethracs) [inline actionlist/app/components/SideMenu.js]: Remove this eslint exception

Comment (PrashanthKumarM) [reply actionlist/app/components/SideMenu.js]: fixed!!

**Review by Nethracs — APPROVED** [review]

**Review by burnumd — CHANGES_REQUESTED** [review]
This PR looks good. If you would, please, just extract the repeated`<View style={style.header}><Text style={style.headerText}>{/* text */}</Text></View>` segments into a `Header` component so they can be reused consistently.

File: actionlist/app/components/contentHeader.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    fontSize: 10,
+    color: Colors.IUCrimson,
+    fontFamily: 'BentonSansBold, Arial, sans-serif',
+    margin: 10,
+  },
+});
+
+const ContentHeader = ({ title }) => (
+  <View>
+    <Text style={style.headerText}>{title}</Text>
```
Comment (burnumd) [inline actionlist/app/components/contentHeader.js]: I'd recommend not using a `title` prop and just use the `children` prop. It looks like it's allowed for `Text` components to have node children, so if you just use children you can pass off prop validation to the react-native component and allow yourself more flexibility.

**Review by burnumd — CHANGES_REQUESTED** [review]

File: actionlist/app/components/contentHeader.js (hunk 2 above)
Comment (PrashanthKumarM) [reply actionlist/app/components/contentHeader.js]: I tried to use children and lint threw an error saying I need to have a prop validation for it

Comment (burnumd) [reply actionlist/app/components/contentHeader.js]: Yes, the type of `children` should be documented as `React.PropTypes.node`: https://github.com/yannickcr/eslint-plugin-react/issues/7

**Review by burnumd — APPROVED** [review]

**Pull request merged**